compiler/statements.go

```diff
@@ -126,6 +126,9 @@ func (c *funcContext) translateStmt(stmt ast.Stmt, label *types.Label) {
 				if _, isInterface := implicit.Type().Underlying().(*types.Interface); !isInterface {
 					value += ".$val"
 				}
+				if typesutil.IsJsObject(implicit.Type().Underlying()) {
+					value += ".object"
+				}
```

Comment:
Would everything still work if the if statement is moved into the one above?

In other words, can ".object" be added without ".$val" in front of it?

Asking to learn.

Reply:
> If you mean:
> 
> ```go
> if _, isInterface := implicit.Type().Underlying().(*types.Interface); !isInterface {
>   value += ".$val"
> 
>   if typesutil.IsJsObject(implicit.Type().Underlying()) {
>     value += ".object"
>   }
> }
> ```

Yes, that's what I meant. I wanted to know if something would break/change if the code were written that way.

> then yes, that's logically equivalent (because something cannot have an underlying type that is both an interface and `*js.Object`).

Okay, I see.

To confirm my understanding is correct, let me answer my question:

> In other words, can ".object" be added without ".$val" in front of it?

From my understanding of what you said, no, ".object" cannot be added without ".$val" also being added.

Only these scenarios are possible:

- `implicit.Type().Underlying()` type is an interface - then `value` stays as is
- `implicit.Type().Underlying()` type is not an interface, but it's not `*js.Object` either - then `value` gets ".$val" appended
- `implicit.Type().Underlying()` type is not an interface, it is `*js.Object` - then `value` gets ".$val.object" appended

If my understanding is indeed correct, then wouldn't the code become simpler (to understand and reason about) if you do move the if statement inside?

Reply:
Actually, I think the readability/clarity of can happen would be improved even more if you made it an if/else if statement with these 2 cases:

```Go
if typesutil.IsJsObject(implicit.Type().Underlying()) {
	value += ".$val.object"
} else if _, ok := implicit.Type().Underlying().(*types.Interface); !ok {
	value += ".$val"
}
```

(I also renamed `isInterface` to just `ok` because I think the long variable name doesn't actually improve readability here; but this is orthogonal and optional.)

What do you think? Is that better, worse, or the same in your opinion?

Reply:
I easy either way. I don't think it's particularly unclear one way or the other. I've made the change, rebased this branch and pushed.